### PR TITLE
[ fix ] Make spaces in args work correctly in `pack run`

### PR DESCRIPTION
**Pull request by buzden**
If you run your project with `pack run <whatever> ...` and have an argument with spaces in it, e.g. `pack run <whatever> foo "bar baz"`, now these arguments are wrongly passed as three arguments `foo`, `bar` and `baz` instead of two. This PR is the simplest fix I can imagine.

I used [a tiny example](https://github.com/buzden/idris-playground/tree/master/print-args) as a local test with running `pack run print-args a b "c d"`.

### CODE REVIEW

**Comment by buzden** [comment]
Well, surely, this fix only pushes non-working case to be more sophisticated. Say, `pack run print-args a b "c' 'd"` does not work correctly... Thus, we need some less trivial fix.

**Comment by stefan-hoeck** [comment]
I'll do some experimentation with this and will then report back here.

**Comment by buzden** [comment]
I completely reworked my solution of this problem by switching to escaped versions of system command runners and reusing escaping mechanism from `base`. This lead to replacing `String` representation of commands to `List String`.

However, I changed the status of this PR to draft because there are some places where simple approach does not work. Couple of commands significantly use shell special characters like `>`, `<` and `*`, which are escaped by the standard functions as well. I need to think how to work around this.

BTW, as far as I can see, one place where `>` is used, this operator can be simply removed, it is for `katla` call; as far as I investigated, `katla` supports output file as the third file argument.

**Comment by buzden** [comment]
Well, I've managed to make able to not to escape selected parts of a command, but in a cost of increased `%ambiguity_depth`, which should be set to be maximum length of arguments in a command in a module. Let's see if I can beat it (however, I doubt).

**Comment by buzden** [comment]
Well, I've managed to do this without increasing ambiguity depth. I've checked my original example and it works. But this change anyway requires additional checks that I didn't screw some commands unintentionally. I will check myself later, but for some time I definitely will have my eyes blurry.

File: src/Pack/Core/Types.idr [hunk 1]
```diff
@@ -407,6 +407,74 @@ export
 dependencies : Desc t -> List PkgName
 dependencies d = dependencies d.desc
 
+--------------------------------------------------------------------------------
+--          CmdArg
+--------------------------------------------------------------------------------
+
```
Comment (stefan-hoeck) [inline src/Pack/Core/Types.idr:413]: Please add some docstrings to these new types and functions: What do they represent, why are they necessary, how should the be used.

**Review by stefan-hoeck — CHANGES_REQUESTED** [review]
This looks good and is almost ready to merge. Please add some docstrings to the new data types in an attempt to keep the source code well documented.

File: src/Pack/Core/Types.idr (hunk 1 above)
Comment (buzden) [reply src/Pack/Core/Types.idr:413]: Done

**Review by stefan-hoeck — APPROVED** [review]

**Comment by stefan-hoeck** [comment]
Thank you so much for your work on this!

**Pull request merged**